Tests/L0/SonarQubePostTest/SummaryReport.ps1

```diff
@@ -53,16 +64,31 @@ Register-Mock IsPrBuild {$false}
 Register-Mock GetTaskContextVariable {$false} -- "MSBuild.SonarQube.IncludeReport" 
 
 # Act
-UploadSummaryMdReport
+CreateAndUploadReport
 
 # Assert
 Assert-WasCalled Write-Host -ArgumentsEvaluator { $args[0].StartsWith("##vso[task.addattachment type=Distributedtask.Core.Summary;name=SonarQube Analysis Report;") }
 
 # Cleanup
 $file.Dispose()
 [IO.Directory]::Delete($tempDir, $true)    
-Unregister-Mock  GetSonarQubeOutDirectory
-Unregister-Mock  Write-Host 
+Unregister-Mock GetSonarQubeOutDirectory
+Unregister-Mock Write-Host 
 Unregister-Mock IsPrBuild   
 
 
+### Test 3 - Fetching the quality gate warnings and errors
```

Comment:
I don't see the value in having tests that compare HTML so I added a unit test for the logic that forms the "view model"